# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/EGLSync.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/EGLSync.h
--- a/gfx/angle/checkout/src/libANGLE/EGLSync.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/EGLSync.h	Sat Sep 07 13:38:36 2019 +0000
@@ -22,29 +22,47 @@
 class EGLSyncImpl;
 }  // namespace rx
 
+namespace gl
+{
+class Context;
+}  // namespace gl
+
 namespace egl
 {
-class Sync final : public angle::RefCountObject<Display, angle::Result>
+class Sync final : public angle::RefCountObject<Display, angle::Result>, public LabeledObject
 {
   public:
     Sync(rx::EGLImplFactory *factory, EGLenum type, const AttributeMap &attribs);
     ~Sync() override;
 
+    void setLabel(EGLLabelKHR label) override;
+    EGLLabelKHR getLabel() const override;
+
     void onDestroy(const Display *display) override;
 
-    Error initialize(const Display *display);
-    Error clientWait(const Display *display, EGLint flags, EGLTime timeout, EGLint *outResult);
-    Error serverWait(const Display *display, EGLint flags);
+    Error initialize(const Display *display, const gl::Context *context);
+    Error clientWait(const Display *display,
+                     const gl::Context *context,
+                     EGLint flags,
+                     EGLTime timeout,
+                     EGLint *outResult);
+    Error serverWait(const Display *display, const gl::Context *context, EGLint flags);
     Error getStatus(const Display *display, EGLint *outStatus) const;
 
+    Error dupNativeFenceFD(const Display *display, EGLint *result) const;
+
     EGLenum getType() const { return mType; }
     EGLint getCondition() const { return mCondition; }
+    EGLint getNativeFenceFD() const { return mNativeFenceFD; }
 
   private:
     std::unique_ptr<rx::EGLSyncImpl> mFence;
 
+    EGLLabelKHR mLabel;
+
     EGLenum mType;
     static constexpr EGLint mCondition = EGL_SYNC_PRIOR_COMMANDS_COMPLETE_KHR;
+    EGLint mNativeFenceFD;
 };
 
 }  // namespace egl